### ISSUE TITLE
Cannot use a signing certificate

### ISSUE THREAD

**Issue by gmajka** [PowerShell/Win32-OpenSSH#1423]
**"OpenSSH for Windows" version**
7.9.0.0
 
**Server OperatingSystem**
Windows Server 2008 R2 Standard

**Client OperatingSystem**
Red Hat Enterprise Linux Server release 7.6 (Maipo)

**What is failing**
I configured public key authentication between machines. I can successfully log in and run every command from linux machine, e.g.
$  ssh user1@host whoami
host\user1

but I am not able to digital sign an executable using a certificate that was installed in a Windows certificate store for a current user (user1 in this example).

Is it possible that when logged using openssh the certificate store for current user is not available?
I found this information in the docs:
"A remote session opened via key based authentication does not have associated user credentials and hence is not capable of outbound authentication as the user. This is by design and goes by the rules of standard Windows security."
(https://github.com/PowerShell/Win32-OpenSSH/wiki/SSH-remote-sessions-on-Windows)
on the other hand I am able to check that I am logged as the right user (whoami returning correct host\user1 information).

Is there a way to have access to certificates installed for a current user over ssh session?

Note: I tested a scenario in which a certificate is stored per Local Computer (not current user) and it works after adding user1 to private key permissions, but for a reason I cannot use the certificates from Local Computer store in a final solution for digital signing.

**Expected output**
$ ssh user1@host  "\"C:/Program Files (x86)/Windows Kits/10/bin/10.0.16299.0/x86/signtool.exe\" sign /a /v /t \"http://timestamp.verisign.com/scripts/timestamp.dll\" /sm /debug c:\temp\zip.exe"

The following certificates were considered:
    Issued to: My company
    Issued by: Symantec Class 3 SHA256 Code Signing CA
    Expires:   Tue Jul 30 01:59:59 2019
    SHA1 hash: 94FF.....................omitted

After EKU filter, 1 certs were left.
After expiry filter, 1 certs were left.
After Private Key filter, 1 certs were left.
The following certificate was selected:
    Issued to: My company
    Issued by: Symantec Class 3 SHA256 Code Signing CA
    Expires:   Tue Jul 30 01:59:59 2019
    SHA1 hash: 94FF.....................omitted

Done Adding Additional Store
Successfully signed: c:\temp\zip.exe

Number of files successfully Signed: 1
Number of warnings: 0
Number of errors: 0

**Actual output**
$ ssh user1@host  "\"C:/Program Files (x86)/Windows Kits/10/bin/10.0.16299.0/x86/signtool.exe\" sign /a /v /t \"http://timestamp.verisign.com/scripts/timestamp.dll\" /debug c:\temp\zip.exe"

The following certificates were considered:
    Issued to: My company
    Issued by: Symantec Class 3 SHA256 Code Signing CA
    Expires:   Tue Jul 30 01:59:59 2019
    SHA1 hash: 94FF.....................omitted

After EKU filter, 1 certs were left.
After expiry filter, 1 certs were left.
After Private Key filter, 0 certs were left.
SignTool Error: No certificates were found that met all the given criteria.

**Comment by jborean93** [comment]
> Is there a way to have access to certificates installed for a current user over ssh session?

You need to log on with password auth. Without knowing the user's password you cannot unlock the credential store for that user and use things like private keys. You can technically still use ssh key auth and then log on again with credentials through that SSH session but because you still need the password you should just use password auth.

**Comment by gmajka** [comment]
Hi Jordan,
Thanks for the answer. In the meantime the certificate I used for signing expired and we obtained a new certificate in another way (a new more secure process is in place in my company) and surprisingly the new one works fine. It's still imported in Windows certificate store per current user and not per machine. This time signtool is able to get the certificate via ssh connection and this is still key based authentication method (not user/password method).
I am not sure what actually changed. I think the only change is how the certificate was imported to the Windows certificate store. Previously I did it manually from pfx file and this time it was done automatically by some tool provided from a signing team (I didn't have access to pfx file).
